rewrite-maven/src/main/java/org/openrewrite/maven/AddDependency.java

```diff
@@ -235,7 +238,14 @@ public Xml visitDocument(Xml.Document document, ExecutionContext ctx) {
                     }
                 }
 
-                if(onlyIfUsing == null && getResolutionResult().getParent() != null && acc.pomsDefinedInCurrentRepository.contains(getResolutionResult().getParent().getPom().getGav())) {
+                if (onlyIfUsing == null && getResolutionResult().getParent() != null && acc.pomsDefinedInCurrentRepository.contains(getResolutionResult().getParent().getPom().getGav())) {
+                    return maven;
+                }
+
+                if (!getResolutionResult().getPom().getSubprojects().isEmpty()
+                    && (getResolutionResult().getModules().isEmpty()
+                        || getResolutionResult().getModules().stream().map(MavenResolutionResult::getPom).map(ResolvedPom::getGav).map(ResolvedGroupArtifactVersion::getArtifactId).noneMatch(art -> getResolutionResult().getPom().getSubprojects().contains(art))
```

Comment:
The bot wants you to place the `&&` and `||` on the previous line, in case that wasn't clear before.

Any why the use of Streams API in one of our most used recipes? Does it help to extract the condition into a properly named method such that you can do an early return from a loop there?